src/test/java/com/google/edith/SearchServletTest.java

```diff
@@ -0,0 +1,241 @@
+    when(response.getWriter()).thenReturn(writer);
+
+    searchServlet.doGet(request, response);
+
+    verify(response, times(1)).getWriter();
+  }
+
+  // Check if receipt response has all fields.
+  @Test
+  public void doGet_whenKindIsReceipt_jsonContainsAllReceiptFields() throws IOException {
+    setUpPostForReceipt();
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(writer);
+
+    searchServlet.doGet(request, response);
+
+    String servletResponse = stringWriter.toString();
+    assertTrue(servletResponse.contains("userId"));
+    assertTrue(servletResponse.contains("storeName"));
+    assertTrue(servletResponse.contains("date"));
+    assertTrue(servletResponse.contains("name"));
+    assertTrue(servletResponse.contains("fileUrl"));
+    assertTrue(servletResponse.contains("price"));
+    assertTrue(servletResponse.contains("items"));
+  }
+
+  // Check if getWriter method is called for Item kind.
+  @Test
+  public void doGet_whenKindIsItem_callsGetWriter() throws IOException {
+    setUpPostForItem();
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(writer);
+    searchServlet.doGet(request, response);
+    verify(response, times(1)).getWriter();
+  }
+
+  // Check if item response has all fields.
+  @Test
+  public void doGet_whenKindIsItem_jsonContainsAllItemFields() throws IOException {
+    setUpPostForItem();
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(writer);
+
+    searchServlet.doGet(request, response);
+
+    String servletResponse = stringWriter.toString();
+    assertTrue(servletResponse.contains("userId"));
+    assertTrue(servletResponse.contains("name"));
+    assertTrue(servletResponse.contains("date"));
+    assertTrue(servletResponse.contains("price"));
+    assertTrue(servletResponse.contains("quantity"));
+    assertTrue(servletResponse.contains("category"));
+    assertTrue(servletResponse.contains("expiration"));
+  }
+
+  // Sets up conditions by populating kind and receipt field of SearchServlet class.
+  private void setUpPostForReceipt() throws IOException {
+    when(request.getParameter("kind")).thenReturn("Receipt");
+    when(request.getParameter("name")).thenReturn("weekend");
+    Entity receiptEntity = new Entity("Receipt");
+    ImmutableList<Entity> entities = ImmutableList.of(receiptEntity);
+    Item item =
+        Item.builder()
+            .setUserId("12345")
+            .setName("apple")
+            .setPrice(1.5)
+            .setQuantity(2)
+            .setDate("date")
+            .setCategory("fruit")
+            .setExpiration("expireDate")
+            .build();
+    // Item item1 = new Item("12345", "apple", 1.5f, 2, "fruit", "date");
```

Comment:
nit: can this be deleted since the builder is being used now

Reply:
Deleted it.